substrate/frame/support/test/tests/enum_deprecation.rs

```diff
@@ -0,0 +1,352 @@
+			#[pallet::compact] foo: u32,
+		) -> DispatchResultWithPostInfo {
+			Self::deposit_event(Event::B);
+			if foo == 0 {
+				Err(Error::<T>::InsufficientProposersBalance)?;
+			}
+
+			Ok(().into())
+		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight({1})]
+		pub fn foo_index_out_of_order(_origin: OriginFor<T>) -> DispatchResult {
+			Ok(())
+		}
+
+		// Test for DispatchResult return type
+		#[pallet::call_index(2)]
+		#[pallet::weight({1})]
+		pub fn foo_no_post_info(_origin: OriginFor<T>) -> DispatchResult {
+			Ok(())
+		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight({1})]
+		pub fn check_for_dispatch_context(_origin: OriginFor<T>) -> DispatchResult {
+			with_context::<(), _>(|_| ()).ok_or_else(|| DispatchError::Unavailable)
+		}
+	}
+
+	#[pallet::error]
+	#[derive(PartialEq, Eq)]
+	pub enum Error<T> {
+		/// error doc comment put in metadata
+		InsufficientProposersBalance,
+		NonExistentStorageValue,
+		Code(u8),
+		#[codec(skip)]
+		Skipped(u128),
+		CompactU8(#[codec(compact)] u8),
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(fn deposit_event)]
+	pub enum Event<T: Config>
+	where
+		T::AccountId: SomeAssociation1 + From<SomeType1>,
+	{
+		#[deprecated = "second"]
+		A,
+		#[deprecated = "first"]
+		#[codec(index = 0)]
+		B,
+	}
+
+	#[pallet::genesis_config]
+	#[derive(frame_support::DefaultNoBound)]
+	pub struct GenesisConfig<T: Config>
+	where
+		T::AccountId: From<SomeType1> + SomeAssociation1 + From<SomeType4>,
+	{
+		#[serde(skip)]
+		_config: sp_std::marker::PhantomData<T>,
+		_myfield: u32,
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> BuildGenesisConfig for GenesisConfig<T>
+	where
+		T::AccountId: From<SomeType1> + SomeAssociation1 + From<SomeType4>,
+	{
+		fn build(&self) {
+			let _ = T::AccountId::from(SomeType1); // Test for where clause
+			let _ = T::AccountId::from(SomeType4); // Test for where clause
+		}
+	}
+
+	#[pallet::origin]
+	#[derive(
+		EqNoBound,
+		RuntimeDebugNoBound,
+		CloneNoBound,
+		PartialEqNoBound,
+		PartialOrdNoBound,
+		OrdNoBound,
+		Encode,
+		Decode,
+		TypeInfo,
+		MaxEncodedLen,
+	)]
+	pub struct Origin<T>(PhantomData<T>);
+}
+
+frame_support::parameter_types!(
+	pub const MyGetParam3: u32 = 12;
+);
+
+#[derive_impl(frame_system::config_preludes::TestDefaultConfig)]
+impl frame_system::Config for Runtime {
+	type BaseCallFilter = frame_support::traits::Everything;
+	type RuntimeOrigin = RuntimeOrigin;
+	type Nonce = u64;
+	type RuntimeCall = RuntimeCall;
+	type Hash = sp_runtime::testing::H256;
+	type Hashing = sp_runtime::traits::BlakeTwo256;
+	type AccountId = u64;
+	type Lookup = sp_runtime::traits::IdentityLookup<Self::AccountId>;
+	type Block = Block;
+	type RuntimeEvent = RuntimeEvent;
+	type BlockWeights = ();
+	type BlockLength = ();
+	type DbWeight = ();
+	type Version = ();
+	type PalletInfo = PalletInfo;
+	type AccountData = ();
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type SS58Prefix = ();
+	type OnSetCode = ();
```

Comment:
All of this could have been removed because of `derive_impl`

This is one good reason to clean up all instances of `derive_impl` on `frame_system`, as evidently people do still copy paste it and are not fully familiar with it :)